icepyx/core/visualization.py

```diff
@@ -159,7 +159,9 @@ def __init__(
             self.bbox = query_obj.spatial.extent
 
         else:
-            mrc_bound = query_obj.spatial.extent.minimum_rotated_rectangle
+            mrc_bound = (
+                query_obj.spatial.extent_as_gdf.geometry.unary_union.minimum_rotated_rectangle.bounds
```

Comment:
Probably want the `shapely.geometry` object rather than the tuple (`minx`, `miny`, `maxx`, `maxy`).
```suggestion
                query_obj.spatial.extent_as_gdf.geometry.unary_union.minimum_rotated_rectangle
```
Alternatively, it might also possible to use `lonmin, lonmax, latmin, latmax = query_obj.spatial.extent_as_gdf.geometry.total_bounds` if you want to refactor the code from L162 to L169.

Reply:
Great catches @weiji14, as usual. Thank you!